Description:
  libf2fs_dlutils.so dlopen libf2fs_fmt.so to provide two functions
  dynamically, but this involves locating the libraries. Since f2fs-tools
  supports multi-arch, the library location is not static, it is easier to
  directly use the functions in the linked library directly.
Author: Kai-Chung Yan <seamlikok@gmail.com>
Last-Update: 2015-08-02
--- a/extras/f2fs_utils/f2fs_dlutils.c
+++ b/extras/f2fs_utils/f2fs_dlutils.c
@@ -37,29 +37,6 @@
 #include <f2fs_format_utils.h>
 #define F2FS_DYN_LIB "libf2fs_fmt_host_dyn.so"
 
-int (*f2fs_format_device_dl)(void);
-void (*f2fs_init_configuration_dl)(struct f2fs_configuration *);
-
-int f2fs_format_device(void) {
-	assert(f2fs_format_device_dl);
-	return f2fs_format_device_dl();
-}
-void f2fs_init_configuration(struct f2fs_configuration *config) {
-	assert(f2fs_init_configuration_dl);
-	f2fs_init_configuration_dl(config);
-}
-
 int dlopenf2fs() {
-	void* f2fs_lib;
-
-	f2fs_lib = dlopen(F2FS_DYN_LIB, RTLD_NOW);
-	if (!f2fs_lib) {
-		return -1;
-	}
-	f2fs_format_device_dl = dlsym(f2fs_lib, "f2fs_format_device");
-	f2fs_init_configuration_dl = dlsym(f2fs_lib, "f2fs_init_configuration");
-	if (!f2fs_format_device_dl || !f2fs_init_configuration_dl) {
-		return -1;
-	}
 	return 0;
 }
